store/tikv/oracle/oracles/pd.go

```diff
@@ -28,30 +29,55 @@ const slowDist = 30 * time.Millisecond
 
 // pdOracle is an Oracle that uses a placement driver client as source.
 type pdOracle struct {
-	c pd.Client
+	c  pd.Client
+	mu struct {
+		sync.RWMutex
+		lastTs uint64
+	}
+	quit    chan struct{}
+	updated chan struct{}
 }
 
 // NewPdOracle create an Oracle that uses a pd client source.
 // Refer https://github.com/pingcap/pd/blob/master/pd-client/client.go for more details.
-func NewPdOracle(pdClient pd.Client) oracle.Oracle {
-	return &pdOracle{
-		c: pdClient,
+func NewPdOracle(pdClient pd.Client, updateInterval time.Duration) (oracle.Oracle, error) {
+	o := &pdOracle{
+		c:       pdClient,
+		quit:    make(chan struct{}),
+		updated: make(chan struct{}),
+	}
+	go o.updateTS(updateInterval)
+	// Initialize lastTS by Get.
+	_, err := o.GetTimestamp()
+	if err != nil {
+		o.Close()
+		return nil, errors.Trace(err)
 	}
+	return o, nil
 }
 
 // IsExpired returns whether lockTs+TTL is expired, both are ms.
-func (t *pdOracle) IsExpired(lockTs, TTL uint64) (bool, error) {
-	physical, _, err := t.c.GetTS()
+func (o *pdOracle) IsExpired(lockTs, TTL uint64) (bool, error) {
```

Comment:
Add comment explain that this function may return false negative result as lockTs is compared to `lastTs`.